espnet2/asr/frontend/s3prl.py

```diff
@@ -86,10 +86,10 @@ def _get_upstream(self, frontend_conf):
 
```

Comment:
Is it related to this PR?
If it takes time to review this PR, we may make this part a separate PR.

Reply:
> * do we need to have a special inference code for enhanced speech translation and recognition? I thought that we may just need to have them for training since the inference interface (given a speech to output text) would not be changed w/ and w/o enhancement.

The difference between `enh_s2t_inference` and `asr_inference` is that `enh_s2t_inference` can dump both the enhanced wavs and the text in the inference. I can remove the inference script if this is not necessary.



> Is it related to this PR? If it takes time to review this PR, we may make this part a separate PR.

In some sense, yes. We would like to backpropagate the gradient from ASR to Enh models.